src/test/regress/sql/alter_table_add_column.sql

```diff
@@ -0,0 +1,71 @@
+CREATE SCHEMA alter_table_add_column;
+SET search_path TO alter_table_add_column;
+
+SET citus.next_shard_id TO 1830000;
+SET citus.shard_replication_factor TO 1;
+
+SET client_min_messages TO NOTICE;
+
+CREATE TABLE referenced (int_col integer PRIMARY KEY);
+CREATE TABLE referencing (text_col text);
+SELECT create_distributed_table('referenced', null);
+SELECT create_distributed_table('referencing', null);
+
+CREATE OR REPLACE FUNCTION my_random(numeric)
+  RETURNS numeric AS
+$$
+BEGIN
+  RETURN 7 * $1;
+END;
+$$
+LANGUAGE plpgsql IMMUTABLE;
+
+CREATE COLLATION caseinsensitive (
+	provider = icu,
+	locale = 'und-u-ks-level2'
+);
+
+CREATE TYPE "simple_!\'custom_type" AS (a integer, b integer);
+
+ALTER TABLE referencing ADD COLUMN test_1 integer DEFAULT (my_random(7) + random() + 5) NOT NULL CONSTRAINT fkey REFERENCES referenced(int_col) ON UPDATE SET DEFAULT ON DELETE CASCADE DEFERRABLE INITIALLY DEFERRED;
+ALTER TABLE referencing ADD COLUMN test_2 integer UNIQUE REFERENCES referenced(int_col) ON UPDATE CASCADE ON DELETE SET DEFAULT NOT DEFERRABLE INITIALLY IMMEDIATE;
+ALTER TABLE referencing ADD COLUMN test_3 integer GENERATED ALWAYS AS (test_1 * my_random(1)) STORED UNIQUE REFERENCES referenced(int_col) MATCH FULL;
+ALTER TABLE referencing ADD COLUMN test_4 integer PRIMARY KEY WITH (fillfactor=70) NOT NULL REFERENCES referenced(int_col) MATCH SIMPLE ON UPDATE CASCADE ON DELETE SET DEFAULT;
+ALTER TABLE referencing ADD COLUMN test_5 integer CONSTRAINT unique_c UNIQUE WITH (fillfactor=50);
+ALTER TABLE referencing ADD COLUMN test_6 text COMPRESSION pglz COLLATE caseinsensitive NOT NULL;
+ALTER TABLE referencing ADD COLUMN "test_\'!7" "simple_!\'custom_type";
+
+-- we give up deparsing ALTER TABLE command if it needs to create a check constraint, and we fallback to legacy behavior
+ALTER TABLE referencing ADD COLUMN test_8 integer CHECK (test_8 > 0);
+ALTER TABLE referencing ADD COLUMN test_8 integer CONSTRAINT check_test_8 CHECK (test_8 > 0);
+
```

Comment:
I couldn't see any issues, but it would be good to have a more complex GENERATED ALWAYS AS expressions (e.g. stuff from different schemas)